pkg/client/custom_reader.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"fmt"
+	monv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/prometheus"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientv1 "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CustomReader provides capability to identify client at runtime
+type CustomReader struct {
+	Client *Client
+}
+
+// CustomReaderC provide reader reference for given client
+func CustomReaderC(cli *Client) clientv1.Reader {
+	return &CustomReader{
+		Client: cli,
+	}
+}
+
+// List retrieves list of objects for a given namespace and list options.
+func (r *CustomReader) List(ctx context.Context, list runtime.Object, opts ...clientv1.ListOption) error {
```

Comment:
Since you created the `resources` functions, I think this method should move to there as well. Just change the name to something more meaningful to not clash with the `List` we already have. Are you passing the `opts` arguments when calling this function? If not, please remove.

Reply:
I had removed custom_reader/custer_writer files